Bugsnag/Payload/BugsnagStackframe.m

```diff
@@ -66,6 +70,90 @@ + (BugsnagStackframe *)frameFromDict:(NSDictionary *)dict
     }
 }
 
++ (NSArray<BugsnagStackframe *> *)stackframesWithCallStackSymbols:(NSArray<NSString *> *)callStackSymbols {
+    NSString *pattern = (@"^(\\d+)"             // Capture the leading frame number
+                         @" +"                  // Skip whitespace
+                         @"(\\S+)"              // Image name
+                         @" +"                  // Skip whitespace
+                         @"(0x[0-9a-fA-F]+)"    // Capture the frame address
+                         @"("                   // Start optional group
+                         @" "                   // Skip whitespace
+                         @"(.+)"                // Capture symbol name
+                         @" \\+ "               // Skip " + "
+                         @"\\d+"                // Instruction offset
+                         @")?$"                 // End optional group
+                         );
+    
+    NSError *error;
+    NSRegularExpression *regex =
+    [NSRegularExpression regularExpressionWithPattern:pattern options:0 error:&error];
+    if (!regex) {
+        bsg_log_err(@"%@", error);
+        return nil;
+    }
+    
+    NSMutableArray<BugsnagStackframe *> *frames = [NSMutableArray array];
+    
+    for (NSString *string in callStackSymbols) {
+        NSTextCheckingResult *match = [regex firstMatchInString:string options:0 range:NSMakeRange(0, string.length)];
+        if (match.numberOfRanges != 6) {
+            continue;
+        }
+        NSString *frameNumber = [string substringWithRange:[match rangeAtIndex:1]];
+        NSString *imageName = [string substringWithRange:[match rangeAtIndex:2]];
+        NSString *frameAddress = [string substringWithRange:[match rangeAtIndex:3]];
+        NSRange symbolNameRange = [match rangeAtIndex:5];
+        NSString *symbolName = nil;
+        if (symbolNameRange.location != NSNotFound) {
+            symbolName = [string substringWithRange:symbolNameRange];
+        }
```

Comment:
`symbolName` should fall back to `frameAddress` when there is no match. It can also be asserted in the dummy symbols test.

Reply:
Fixed in 3cd17800795c4be1d1b9938f5c81ede997cfd3df